_inc/lib/widgets.php

```diff
@@ -458,7 +478,6 @@ public static function update_widget( $widget_id, $sidebar, $position, $settings
 		if ( ! isset( $widgets_in_sidebar ) ) {
 			return new WP_Error( 'invalid_data', 'No such sidebar exists', 400 );
 		}
-		self::move_widget_to_sidebar( $widget, $sidebar, $position );
```

Comment:
This one was useful in case we're moving an existing widget to another sidebar - could you please clarify why we're removing it?

Reply:
Thanks for reviewing @tyxla! 🙇‍♀️ 
The way I understand this logic: if widget passed to `update_widget` is not in the `sidebar` 
```
$widgets_in_sidebar = self::get_widgets_in_sidebar( $sidebar );
if ( ! isset( $widgets_in_sidebar ) ) {
 return new WP_Error( 'invalid_data', 'No such sidebar exists', 400 );
}``` 
will return and we would not get to that call anyway. Wouldn't it be clearer to create a separate function for `move_widget` for those cases?

Let me know if I misunderstood your question :)

Reply:
Let me clarify further. The `update_widget` method is a kind of generic and supports many situations. One of the cases is where the widget is currently in sidebar "x" and you want to move it to sidebar "y" while updating it. Another case is when we want to change its position. In both of these cases, `self::move_widget_to_sidebar` would move the widget accordingly, but now that you removed that method, it'll no longer handle those cases. Makes sense?

Reply:
Fwiw, that error message `No such sidebar exists` is not quite precise for what it checks 🤔

Reply:
Good call, worth improving I guess....

Reply:
☝️ right, agreed. For a moment I read the `$widgets_in_sidebar = self::get_widgets_in_sidebar( $sidebar );` as a check whether that specific widget is present.
Two points though:
- the above shows another edge case: if a sidebar is empty and we want to place "our" widget in it, it will return before that happens (`if ( ! isset( $widgets_in_sidebar ) ) `). Is this by design?
- how about we add the line only if it is needed? ( after checking whether the widget is in the sidebar or not)

Reply:
Makes sense to me, maybe it's best to confirm with @eliorivero who wrote the initial code there.

Reply:
your thoughts on it would be appreciated @eliorivero  :) Thanks in advance!

Reply:
I added the logic mentioned above and updated the error message. 
I still do not see the why behind returning on sidebar with no widgets. It is not a blocker at the moment though.